### PR TITLE
[FEATURES] Filtration, Search, Seasons Accordion, Data Service, IoC - NavConfig with apiCallConfig

### PR DESCRIPTION
**Pull request by dimas0211**
created #4:
1. filtration by year and/or genre
2. search functionality
3. ioc pre setup

### CODE REVIEW

File: src/components/Header/Header.container.js [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
 import connectWithViewport from '../../services/connectWithViewport';
 
-const mapStateToProps = () => ({
+import Header from './Header';
+import { setSearchQuery, setSearchMode } from '../../actions/searchMovies';
+import connectWithIoC from '../../services/connectWithIoC';
+import { getMoviesError, getMoviesSuccess } from '../../actions/actionsAPIcall';
+
+const mapStateToProps = ({ ApiReducer, setSearchModeReducer }) => ({
+  error: ApiReducer.error,
```
Comment (AlyonaPianykh) [inline src/components/Header/Header.container.js]: destruct ApiReducer

File: src/components/Header/Header.container.js [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
-import { compose } from 'redux';
-import Header from './Header';
+import { bindActionCreators, compose } from 'redux';
 import connectWithViewport from '../../services/connectWithViewport';
 
-const mapStateToProps = () => ({
+import Header from './Header';
+import { setSearchQuery, setSearchMode } from '../../actions/searchMovies';
+import connectWithIoC from '../../services/connectWithIoC';
+import { getMoviesError, getMoviesSuccess } from '../../actions/actionsAPIcall';
```
Comment (AlyonaPianykh) [inline src/components/Header/Header.container.js]: Also, You can import action like this
```
import * as movieActions from '../../actions/actionsAPIcall';
```
and then use them in bindActionCreators like this
```
const mapDispatchToProps = (dispatch) => bindActionCreators({
 ...movieActions
}, dispatch);
```

Comment (AlyonaPianykh) [reply src/components/Header/Header.container.js]: But it's just for future using, no need to change it now

File: src/components/Header/Header.js [hunk 3]
```diff
@@ -1,24 +1,42 @@
...
+  constructor(props) {
+    super(props);
+
+    autoBind(this);
+  }
+
+  handleSearch(query) {
+    const { setSearchMode, setSearchQuery, history, routingConfig: { search } } = this.props;
+
+    setSearchMode && setSearchMode();
```
Comment (AlyonaPianykh) [inline src/components/Header/Header.js]: Is that 2 api calls here?

File: src/components/MovieItem/MovieItem.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+              {renderGenres()}
+            </span>
+          </p>
+          <p className={`${CN}__info-item-container`}>
+            <span className={`${CN}__info-item-name`}>Release Date: </span>
+            <span className={`${CN}__info-item-value release-year`}>
+              {releaseDate}
+            </span>
+          </p>
+          <p className={`${CN}__info-item-container`}>
```
Comment (AlyonaPianykh) [inline src/components/MovieItem/MovieItem.js]: These all the same jsx constructions but with different name and content and classname. create a separate function that would be returning this jsx with dynamic parameters. no need to copy your own code

File: src/components/MovieItemPage/MovieItemPage.scss [hunk 5]
```diff
@@ -6,8 +6,9 @@
     margin: 25px 15px;
     max-width: 100%;
     padding: 20px;
-    border: 1px solid $border-color;
+    //border: 1px solid $border-color;
```
Comment (AlyonaPianykh) [inline src/components/MovieItemPage/MovieItemPage.scss]: please remove commented line

File: src/components/MovieItemPage/MovieItemPage.scss [hunk 6]
```diff
@@ -87,13 +88,35 @@
...
+    display: inline-block;
+    margin: 20px;
+  }
+
+  &__season-info-container-mobile {
+    width: 100%;
+  }
+
+  &__seasons-container {
+    overflow-x: scroll;
```
Comment (AlyonaPianykh) [inline src/components/MovieItemPage/MovieItemPage.scss]: Why scroll and not auto?

File: src/components/MovieItemPage/MovieItemPage.scss [hunk 7]
```diff
@@ -103,7 +126,38 @@
...
+  }
+
+  &__seasons-with-video-container-large-screen {
+    @include flex($direction: row);
+    height: 740px;
+  }
+
+  &__video-tabs-wrapper-show {
+    width: 55%;
+
```
Comment (AlyonaPianykh) [inline src/components/MovieItemPage/MovieItemPage.scss]: please delete empty line

File: src/components/MovieListPage/MovieListPage.scss [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+.movie-list-page-with-filters {
+  padding-top: 50px;
+  &__tablet {
+    padding-top: 50px;
+  }
+
+  &__mobile {
+    padding-top: 85px;
+  }
+
```
Comment (AlyonaPianykh) [inline src/components/MovieListPage/MovieListPage.scss]: Please remove this empty line

File: src/components/MovieListPage/MovieListPage.scss [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+
+
```
Comment (AlyonaPianykh) [inline src/components/MovieListPage/MovieListPage.scss]: and these

File: src/components/MovieListPagination/MovieListPagination.js [hunk 10]
```diff
@@ -36,17 +36,36 @@ class MovieListPagination extends Component {
...
+      query,
+      movieOrShow,
+      pageParams,
+      filtrationQueryParams,
+      getMoviesSuccess,
+      getMoviesError,
+      apiCallConfig: { movie }
+    } = this.props;
+
+    searchMode === true
```
Comment (AlyonaPianykh) [inline src/components/MovieListPagination/MovieListPagination.js]: why do you need `=== true`?

Comment (AlyonaPianykh) [reply src/components/MovieListPagination/MovieListPagination.js]: you can simplify this code a lot

```
let action = searchMode ? 'searchMovies' : 'getMovieList';
let args = searchMode ? [page, movie, query] : [page, movieOrShow, pageParams, ...filtrationQueryParams]
getDataService[action](...args)     
        .then((data) => getMoviesSuccess(data))
        .catch((error) => getMoviesError(error));

```

File: src/components/MovieMainItemCard/MovieMainItemCard.js [hunk 11]
```diff
@@ -45,21 +61,19 @@ class MovieMainItemCard extends Component {
...
-    } = this.props;
-    const isMobile = device === MOBILE;
+  const onClickLoadMovieData = () => {
+    setOneMovieType(movieOrShow);
+  };
 
-    return (
-      <div className={cx(`${CN}__wrapper`, isMobile && `${CN}__wrapper-mobile`)}>
+  return (
+    <Link className={cx(isLargeScreen && `${CN}__wrapper-large-screen`, 'page-link')} to={`${movieOrShow}${view}`.replace(':id', id)}>
```
Comment (AlyonaPianykh) [inline src/components/MovieMainItemCard/MovieMainItemCard.js]: Please see https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/generatePath.md

File: src/components/Navigation/Navigation.container.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  movieList: ApiReducer.movieListInfo,
+  genres: ApiReducer.genresList.genres,
+  filtrationQueryParams: setFiltrationParamsReducer.filtrationQueryParams
+});
+
+export const NavigationContainer = compose(
+  connectWithViewport(),
+  connect(
+    mapStateToProps,
+    null
```
Comment (AlyonaPianykh) [inline src/components/Navigation/Navigation.container.js]: you don't need to pass null here

File: src/components/Navigation/NavigationMobile/NavigationMob.js [hunk 13]
```diff
@@ -9,28 +9,57 @@ import {
   Divider,
   Button,
   Drawer,
-  List
+  List, withStyles
```
Comment (AlyonaPianykh) [inline src/components/Navigation/NavigationMobile/NavigationMob.js]: on the next line

File: src/components/Routes/Routes.js [hunk 14]
```diff
@@ -3,31 +3,43 @@ import { Switch, Route } from 'react-router-dom';
...
+        to={notFound}
       />
     </Switch>
-  )}
-  />
-);
+  );
+
+  return (
+    <Route render={() => renderPages()} />
```
Comment (AlyonaPianykh) [inline src/components/Routes/Routes.js]: where is path for route? are you sure you need to wrap it in route

File: src/components/Header/Header.container.js (hunk 2 above)
Comment (dimas0211) [reply src/components/Header/Header.container.js]: nice tip, thanks!

File: src/components/Header/Header.js (hunk 3 above)
Comment (dimas0211) [reply src/components/Header/Header.js]: no.
it just sets parameters that i use for ui.
first works just as flag.
second sets query to state for making an following API call
image

File: src/components/MovieListPagination/MovieListPagination.js (hunk 10 above)
Comment (dimas0211) [reply src/components/MovieListPagination/MovieListPagination.js]: oh yes, thank you :)

File: src/components/Routes/Routes.js (hunk 14 above)
Comment (dimas0211) [reply src/components/Routes/Routes.js]: fixed:
image

**Pull request merged**